packages/truffle-core/lib/console.js

```diff
@@ -42,7 +42,7 @@ class Console extends EventEmitter {
     this.repl.on("exit", () => this.emit("exit"));
   }
 
-  start(callback) {
+  async start(callback) {
```

Comment:
Heads up: it's bad practice to make a `callback`-style function also `async`. Doing this has broken things in strange/unexpected ways.

Instead, I think you have to wrap the call to `this.repl.start` inside a `this.web3.eth.getAccounts().then(...)`.